starter/src/main/java/io/javaoperatorsdk/operator/springboot/starter/CrdUploader.java

```diff
@@ -0,0 +1,91 @@
+package io.javaoperatorsdk.operator.springboot.starter;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.function.UnaryOperator;
+
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.KubernetesClient;
+
+import static java.util.Arrays.stream;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+
+public class CrdUploader {
+
+  private static final Logger log = LoggerFactory.getLogger(CrdUploader.class);
+
+  private static final int CRD_READY_WAIT = 2000;
+
+  private final CrdTransformer crdTransformer;
+  private final KubernetesClient kubernetesClient;
+  private final boolean applyOnStartup;
+  private final String crdSuffix;
+  private final String crdPath;
+
+  public CrdUploader(KubernetesClient kubernetesClient, List<CrdTransformer> transformers,
+      boolean applyOnStartup, String crdPath, String crdSuffix) {
+    this.crdTransformer = CrdTransformer.reduce(transformers);
+    this.kubernetesClient = kubernetesClient;
+    this.applyOnStartup = applyOnStartup;
+    this.crdSuffix = crdSuffix;
+    this.crdPath = crdPath;
+  }
+
+  public void upload() throws Exception {
+    if (applyOnStartup) {
+      log.debug("Uploading CRDs with suffix {} under {}", crdSuffix, crdPath);
+      stream(findFiles()).forEach(this::applyCrd);
+    } else {
+      log.debug("Not searching for CRDs to upload");
+    }
+  }
+
+  private File[] findFiles() throws URISyntaxException {
+    var resource = requireNonNull(
+        getClass().getResource(crdPath),
+        "Could not find the configured CRD path");
+
+    return new File(resource.toURI()).listFiles((ignored, name) -> name.endsWith(crdSuffix));
+  }
+
+  private void applyCrd(File crdFile) {
+    try (var is = new FileInputStream(crdFile)) {
+      var crds = kubernetesClient.load(is).items().stream().map(crdTransformer).toList();
+      kubernetesClient.resourceList(crds).createOrReplace();
+
+      Thread.sleep(CRD_READY_WAIT); // readiness is not applicable for CRD, just wait a little
```

Comment:
Wondering if we could do this smarter? Like checking if CRD present periodically - well not sure if that would work. Maybe experiment with it a little?

Reply:
this was actually copied from you commit [here](https://github.com/java-operator-sdk/java-operator-sdk/pull/1079/files#diff-3379770672214df7f1b157f8d722b8a0b9d4017bafb08b3a1dae42d9ee2fef50R67), I didn't bother second guessing you haha

Reply:
Yeah, i know this, is a kinda dummy solution :) 
if we can find something smarter it's better but can stay also like this